File: tutu-backend/src/main/scala/fi/oph/tutu/backend/domain/Hakija.scala
package fi.oph.tutu.backend.domain

case class Hakija(
  henkiloOid: String,
  etunimet: String,
  kutsumanimi: String,
  sukunimi: String,
  kansalaisuus: Seq[Kielistetty],
  hetu: Option[String],
  syntymaaika: String,
  matkapuhelin: Option[String],
  asuinmaa: Kielistetty,
  katuosoite: String,
  postinumero: String,
  postitoimipaikka: String,
  kotikunta: Kielistetty,
  sahkopostiosoite: Option[String],
  yksiloityVTJ: Boolean
)

File: tutu-frontend/src/lib/types/hakija.ts
import { TranslatedName } from '@/src/lib/localization/localizationTypes';

export type Hakija = {
  etunimet: string;
  kutsumanimi: string;
  sukunimi: string;
  kansalaisuus: TranslatedName[];
  hetu?: string;
  syntymaaika: string;
  matkapuhelin?: string;
  asuinmaa: TranslatedName;
  katuosoite: string;
  postinumero: string;
  postitoimipaikka: string;
  kotikunta: TranslatedName;
  sahkopostiosoite?: string;
  yksiloityVTJ: boolean;
};

export type HakijaListItem = {
  etunimet: string;
  sukunimi: string;
};

export const HAKIJA_FIELDS = [
  'etunimet',
  'kutsumanimi',
  'sukunimi',
  'kansalaisuus',
  'hetu',
  'syntymaaika',
  'matkapuhelin',
  'asuinmaa',
  'katuosoite',
  'postinumero',
  'postitoimipaikka',
  'kotikunta',
  'sahkopostiosoite',
];
